### PR TITLE
Add migration guide to version 0.4.0

### PR DESCRIPTION
**Pull request by Ph0enixKM**
This PR introduces:
- New migration guide from 0.3.5 to 0.4.0
- Improved table style to match the website's design
- Fixed issue with different sections between different versions of the documentation not appearing

### CODE REVIEW

**Review by Mte90 — APPROVED** [review]

File: docs/0.4.0-alpha/getting_started/migration_guide.md [hunk 1]
```diff
@@ -0,0 +1,205 @@
+This guide provides a step-by-step walkthrough for migrating code from 0.3.5-alpha to 0.4.0-alpha. The current version introduces several breaking changes. This document outlines the modifications, explains how to adapt your code to maintain the same behavior, and highlights updated features. In this guide we will cover two main categories of changes:
+1.	**Language Features**: Changes and updates to the core language syntax and semantics.
```
Comment (hdwalters) [inline docs/0.4.0-alpha/getting_started/migration_guide.md:2]: Suggest replacing tabs with single spaces.

File: docs/0.4.0-alpha/getting_started/migration_guide.md [hunk 2]
```diff
@@ -0,0 +1,205 @@
...
+
+```ab
+// Before
+unsafe $ ls -a $
+
+// After
+trust $ ls -a $
+```
+
+# Iterator loop
```
Comment (hdwalters) [inline docs/0.4.0-alpha/getting_started/migration_guide.md:19]: Suggest capitalising "Iterator Loop".

File: docs/0.4.0-alpha/getting_started/migration_guide.md [hunk 3]
```diff
@@ -0,0 +1,205 @@
...
+    echo number
+}
+
+// After
+for number in 0..=5 {
+    echo number
+}
+```
+
+# Builtin
```
Comment (hdwalters) [inline docs/0.4.0-alpha/getting_started/migration_guide.md:35]: Suggest "New Builtins".

File: docs/0.4.0-alpha/getting_started/migration_guide.md [hunk 4]
```diff
@@ -0,0 +1,205 @@
...
+import { exit } from "std/env"
+exit(1)
+
+// After
+exit 1
+```
+
+## Lines
+
+Lines was not only moved into a built-in but also it was changed in functionality. Before it accepted a `Text` value which then was splited by new line characters. In this version the newly added builtin accepts a path to file for which it returns each line.
```
Comment (hdwalters) [inline docs/0.4.0-alpha/getting_started/migration_guide.md:69]: Might read better as "which was then split by newline characters".

File: docs/0.4.0-alpha/getting_started/migration_guide.md [hunk 5]
```diff
@@ -0,0 +1,205 @@
+This guide provides a step-by-step walkthrough for migrating code from 0.3.5-alpha to 0.4.0-alpha. The current version introduces several breaking changes. This document outlines the modifications, explains how to adapt your code to maintain the same behavior, and highlights updated features. In this guide we will cover two main categories of changes:
+1.	**Language Features**: Changes and updates to the core language syntax and semantics.
+2.	**Standard Library Updates**: Modifications to existing standard library functions and their usage.
+
+Follow along to ensure a smooth transition to the new version. Let’s get started!
+
+# Command
```
Comment (hdwalters) [inline docs/0.4.0-alpha/getting_started/migration_guide.md:7]: Consider making the heading a bit more explicit, e.g. "Rename Unsafe Keyword".

File: docs/0.4.0-alpha/getting_started/migration_guide.md [hunk 6]
```diff
@@ -0,0 +1,205 @@
...
+echo len([1, 2, 3])
+
+// After
+echo len("Hello there!")
+echo len([1, 2, 3])
+```
+
+## Exit
+
+Exit is used so often that we decided to move it as a separate built-in as well. It's implemented as a builtin statement which means that we can remove the parentheses.
```
Comment (hdwalters) [inline docs/0.4.0-alpha/getting_started/migration_guide.md:56]: Replace "built-in" with "builtin" throughout this file.

File: docs/0.4.0-alpha/getting_started/migration_guide.md [hunk 7]
```diff
@@ -0,0 +1,205 @@
...
+// Before
+import { lines } from "std/text"
+import { file_read } from "std/fs"
+lines(file_read("/path/to/file"))
+
+// After
+lines("/path/to/file")
+```
+
+> DETAILS: This built-in is specifically optimized for direct use as an iterator in a for loop. When used this way, it does not load the entire file into memory but processes it line by line, ensuring efficient resource usage.
```
Comment (hdwalters) [inline docs/0.4.0-alpha/getting_started/migration_guide.md:81]: Consider surrounding `for` with backticks.

File: docs/0.4.0-alpha/getting_started/migration_guide.md [hunk 8]
```diff
@@ -0,0 +1,205 @@
...
+// Before
+import { lines } from "std/text"
+lines(long_text)
+
+// After
+import { split_lines } from "std/text"
+split_lines(long_text)
+```
+
+
```
Comment (hdwalters) [inline docs/0.4.0-alpha/getting_started/migration_guide.md:95]: Unnecessary blank line.

File: docs/0.4.0-alpha/getting_started/migration_guide.md [hunk 9]
```diff
@@ -0,0 +1,205 @@
...
+# Standard Library
+
+In order to keep a consistent standard library function naming, most of the functions have been renamed with one exception.
+
+## Make executable (`make_executable`)
```
Comment (hdwalters) [inline docs/0.4.0-alpha/getting_started/migration_guide.md:100]: Suggest capitalising "Make Executable".

File: docs/0.4.0-alpha/getting_started/migration_guide.md [hunk 10]
```diff
@@ -0,0 +1,205 @@
...
+// Before
+import { make_executable } from "std/fs"
+make_executable("script.ab")
+
+// After
+import { file_chmod } from "std/fs"
+file_chmod("script.ab", "+x")
+```
+
+## Renamed functions
```
Comment (hdwalters) [inline docs/0.4.0-alpha/getting_started/migration_guide.md:114]: Suggest capitalising "Renamed Functions".

File: docs/0.4.0-alpha/getting_started/migration_guide.md [hunk 11]
```diff
@@ -0,0 +1,205 @@
...
+
+New Name             |Old Name
+---------------------|---------------------
+file_download        |download
+
+### `std/math`
+
+New Name             |Old Name
+---------------------|---------------------
+std/maths            |abs
```
Comment (hdwalters) [inline docs/0.4.0-alpha/getting_started/migration_guide.md:184]: Should be `math_abs`.

File: docs/0.4.0-alpha/index.json [hunk 12]
```diff
@@ -4,6 +4,10 @@
             "path": "getting_started/getting_started",
             "title": "Getting Started",
             "docs": [
+                {
+                    "path": "getting_started/migration_guide",
```
Comment (hdwalters) [inline docs/0.4.0-alpha/index.json:8]: Consider reordering these pages; you can't do anything until you've installed Amber anyway, and many users (for whom 0.4.0 will be their first version) will care more about basic usage than migration:
* Installation
* Usage
* Migration Guide

**Review by hdwalters — CHANGES_REQUESTED** [review]
Reviewed only the Markdown changes.

File: docs/0.4.0-alpha/getting_started/migration_guide.md [hunk 13]
```diff
@@ -16,7 +16,7 @@ unsafe $ ls -a $
 trust $ ls -a $
 ```
 
-# Iterator loop
+# Iterator Loop keyword rename
```
Comment (hdwalters) [inline docs/0.4.0-alpha/getting_started/migration_guide.md:19]: Suggest "Rename of `loop` Keyword" for consistency with above.

**Review by hdwalters — APPROVED** [review]

**Review by Mte90 — APPROVED** [review]

**Pull request merged**